cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -77,10 +103,17 @@ public AbstractContext(Program program, RunId runId,
     // todo: this should be instantiated on demand, at run-time dynamically. Esp. bad to do that in ctor...
     // todo: initialized datasets should be managed by DatasetContext (ie. DatasetInstantiator): refactor further
     this.datasets = Datasets.createDatasets(dsInstantiator, datasets, runtimeArguments);
+    this.adapterSpec = adapterSpec;
+    this.pluginInstantiator = pluginInstantiator;
   }
 
   public abstract Metrics getMetrics();
 
+  @Nullable
+  public AdapterSpecification getAdapterSpec() {
```

Comment:
AdapterSpecification is not in cdap-api yet. Hence no for now. We can move it there when we move AdapterSpecification.